functional-tests/src/test/groovy/FrontendCore/spec/Permit.groovy

```diff
@@ -11,17 +16,121 @@ import utils.*
+    }
+
+    def "User can upload a doc to a permit"(){
+        when: "I open the edit initial permit modal."
+        permitTab.permitRow.children().has(text:"Add/Edit").click()
+        permitTab.openFileModalButton.click()
+        
+        and: "Upload a test file to the permit."
+        def uploadedFile = dir.newFile(Const.TEST_FILE_NAME) << Const.TEST_FILE_CONTENT
+        permitTab.uploadField = uploadedFile.absolutePath
+        permitTab.uploadCompleteMessage
+        permitTab.editPermitFileButton.click()
+        downloadTestFileLink
 
-        then:"I should see a list of permit"
-        assert permitTab.permit_no.text() != ''
+        then: "The file is attatched to the permit."
+        downloadTestFileLink.text() == Const.TEST_FILE_NAME
     }
 
+    def "User can download a doc from a permit"(){
+        when: "The user navigates to the test permit's files"
+        permitTab.permitRow.children().has(text:"Add/Edit").click()
+
+        and: "User downloads the  test file"
+        downloadTestFileLink.click()
+        print(Const.DOWNLOAD_PATH+'/'+Const.TEST_FILE_NAME)
+        def file = new File(Const.DOWNLOAD_PATH+'/'+Const.TEST_FILE_NAME)
+        //allow time for the file to be created in the DOWNLOAD_PATH
+        int counter = 0
+        while (!file.exists()) {
+            if (counter>=20 ){
+                throw(new Error("Could not find the file"))
+            }
+            sleep(1000)
+            counter++
+        }
+        String lineString = file.getText('UTF-8')
+        file.delete()
+
+        then: "The doc upload complete message is shown"
+        assert lineString == Const.TEST_FILE_CONTENT
+    }
+
+    def "User can add an amendment to a permit"(){
+        when: "A user opens the Add Permit Amendment Modal"
+        permitTab.permitRow.children().has(text:"Add/Edit").click()
+        moveToFooterAndHoverOnEdit()
+        permitTab.addAmendmentButton.click()
+
+        and: "Completes all the fields"
+        newPermitForm.completePermitAmendment() 
+        permitTab.amendmentDescriptionSpecific
+       
+        then: "An Amendment is added to the permit in question"
+        assert permitTab.amendmentDescriptions.children()[0].text()== "A fancy description"
+    }
+    
+    def "User can add an amalgamate a permit"(){
+        when: "The user amalgamates a permit."
+        moveToFooterAndHoverOnEdit()
+        permitTab.amalgamatePermitButton.click()
+        newPermitForm.amalgamatePermit()
+        waitFor(){permitTab.permitRow.children().has(text:"Add/Edit")}
+
+        and: "The user hovers over the Add/Edit button."
+        moveToFooterAndHoverOnEdit()
+        waitFor() {permitTab.hoverDropdown}
+
+        then: "There is no 'amalgamate' option in the Add/Edit dropdown."
+        assert permitTab.hoverDropdown.children().has(text:"Add permit amendment").isDisplayed() == true 
+        assert permitTab.hoverDropdown.children().has(text:"Amalgamate permit").isDisplayed() == false
```

Comment:
Nice negative test 👍 

Style-wise, you could equally use Groovy truth to assert without a comparator e.g.

```suggestion
        assert !permitTab.hoverDropdown.children().has(text:"Amalgamate permit").isDisplayed()
```

Reply:
much nicer